plugins/modules/ios_evpn_ethernet.py

```diff
@@ -0,0 +1,1013 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright 2025 Red Hat
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+"""
+The module file for ios_evpn_ethernet
+"""
+
+from __future__ import absolute_import, division, print_function
+
+
+__metaclass__ = type
+
+DOCUMENTATION = """
+module: ios_epvn_ethernet
+short_description: Resource module to configure L2VPN EVPN Ethernet Segment.
+description: This module manages the L2VPN EVPN Ethernet Segment attributes of Cisco IOS network devices.
+version_added: 9.2.0
+author:
+  - Sagar Paul (@KB-perByte)
+  - Jorgen Spange (@jorgenspange)
+notes:
+  - Tested against Cisco IOSXE Version 17.6 on CML.
+  - This module works with connection C(network_cli).
+    See U(https://docs.ansible.com/ansible/latest/network/user_guide/platform_ios.html)
+  - The module examples uses callback plugin (callback_result_format=yaml) to generate task
+    output in yaml format.
+options:
+  config:
+    description: A dictionary of L2VPN EVPN Ethernet Segment options
+    type: list
+    elements: dict
+    suboptions:
+      segment:
+        description:
+          - L2VPN EVPN Ethernet Segment, l2vpn evpn ethernet-segment 1
+        type: str
+        required: true
+      default:
+        description: Set a command to its defaults
+        type: bool
+      df_election:
+        description: Designated forwarder election parameters
+        type: dict
+        suboptions:
+          preempt_time:
+            description: Preempt time before advertising routes
+            type: int
+          wait_time:
+            description: Designated forwarder election wait time
+            type: int
+      identifier:
+        description: Ethernet Segment Identifiers
+        type: dict
+        suboptions:
+          identifier_type:
+            description:
+              - Type 0 (arbitrary 9-octet ESI value)
+              - Type 3 (MAC-based ESI value)
+            type: str
+            choices:
+              - '0'
+              - '3'
```

Comment:
any reason of preferring string over int ?

Reply:
The appliance had been notorious for adding a type here and there between releases, and the type has been int and string at times. The model just stays safe.